integration-tests/src/test_loop/utils/network.rs

```diff
@@ -13,7 +14,7 @@ use std::sync::{Arc, Mutex};
 pub fn partial_encoded_chunks_dropper(
     chunks_storage: Arc<Mutex<TestLoopChunksStorage>>,
     epoch_manager_adapter: Arc<dyn EpochManagerAdapter>,
-    validator_of_chunks_to_drop: AccountId,
+    drop_chunks_condition: Box<dyn Fn(ShardChunkHeader, Arc<dyn EpochManagerAdapter>) -> bool>,
```

Comment:
nit: maybe be nicer to type alias `Box<dyn Fn(ShardChunkHeader, Arc<dyn EpochManagerAdapter>) -> bool>` with a short comment explaining what this construct represents